custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/coreprovider_test.go

```diff
@@ -0,0 +1,549 @@
+	timeInfo, metric, err := provider.GetContainerMetrics(pods...)
+	if err != nil {
+		t.Fatalf("Provider error: %s", err)
+	}
+
+	expectedMetrics := [][]metrics.ContainerMetrics{
+		{{"cont2", corev1.ResourceList{"cpu": contCpuRes["namespace2:pod1"]["cont2"], "memory": contRamRes["namespace2:pod1"]["cont2"]}},
+			{"cont1", corev1.ResourceList{"cpu": contCpuRes["namespace2:pod1"]["cont1"], "memory": contRamRes["namespace2:pod1"]["cont1"]}}},
+		{{"cont1", corev1.ResourceList{"cpu": contCpuRes["namespace1:pod1"]["cont1"], "memory": contRamRes["namespace1:pod1"]["cont1"]}},
+			{"cont2", corev1.ResourceList{"cpu": contCpuRes["namespace1:pod1"]["cont2"], "memory": contRamRes["namespace1:pod1"]["cont2"]}}},
+		{{"cont1", corev1.ResourceList{"cpu": contCpuRes["namespace1:pod2"]["cont1"], "memory": contRamRes["namespace1:pod2"]["cont1"]}},
+			{"cont2", corev1.ResourceList{"cpu": contCpuRes["namespace1:pod2"]["cont2"], "memory": contRamRes["namespace1:pod2"]["cont2"]}}},
+	}
+	timeExpected := []api.TimeInfo{
+		{time3, time.Minute},
+		{time1, time.Minute},
+		{time2, time.Minute},
+	}
+
+	if len(pods) != len(metric) {
+		t.Fatalf("Unexpected result. Expected len: \n%v,\n received: \n%v", len(pods), len(metric))
+	}
+	if len(pods) != len(timeInfo) {
+		t.Fatalf("Unexpected result. Expected len: \n%v,\n received: \n%v", len(pods), len(timeInfo))
+	}
+	for i := range pods {
+		if !reflect.DeepEqual(metric[i], expectedMetrics[i]) && !reflect.DeepEqual(metric[i], []metrics.ContainerMetrics{expectedMetrics[i][1], expectedMetrics[i][0]}) {
+			t.Errorf("Unexpected result. Expected: \n%v,\n received: \n%v", expectedMetrics[i], metric[i])
+		}
+
+		if !reflect.DeepEqual(timeInfo[i], timeExpected[i]) {
+			t.Errorf("Unexpected result. Expected: \n%v,\n received: \n%v", timeExpected[i], timeInfo[i])
+		}
+	}
+}
+func TestCoreprovider_GetContainerMetrics_MissingContainer(t *testing.T) {
+	pods := []apitypes.NamespacedName{{"namespace1", "pod1"}}
+	time1, _ := time.Parse(time.RFC3339, "2017-01-02T13:01:00Z")
+
+	contCpuRes := map[string]map[string]resource.Quantity{
+		"namespace1:pod1": {"cont1": fromDouble(0.1), "cont3": fromDouble(0.3)},
+	}
+	contRamRes := map[string]map[string]resource.Quantity{
+		"namespace1:pod1": {"cont1": fromInt(11), "cont2": fromInt(12)},
+	}
+	timeRes := map[string]api.TimeInfo{
+		"namespace1:pod1": {time1, time.Minute},
+	}
+	var provider = CoreProvider{podClient([]string{doubleQuote(pods[0].Name)}, contCpuRes, timeRes, contRamRes, timeRes, t)}
+	timeInfo, metric, err := provider.GetContainerMetrics(pods...)
+	if err != nil {
+		t.Fatalf("Provider error: %s", err)
+	}
+
+	expectedMetrics := [][]metrics.ContainerMetrics{
+		{{"cont1", corev1.ResourceList{"cpu": contCpuRes["namespace1:pod1"]["cont1"], "memory": contRamRes["namespace1:pod1"]["cont1"]}}},
+	}
+	timeExpected := []api.TimeInfo{
+		{time1, time.Minute},
+	}
```

Comment:
Setup for those tests is too big and doesn't really show what's your realy testing